### PR TITLE
Feature 1139 rename parts with email adress

### LINKED ISSUE #1139: Rename parts with `emailAdress`to `emailAddress`
## Situation
Accidently we used `emailAdress` instead of `emailAddress` at some locations

## Wanted
We want that fixed and the typo removed.

## Solution
- [ ]  create `flyway` script which renames the column in table `ADM_USER_SELFREGISTRATION` (+ revert script to wrong name!)
- [ ]  rename  `Signup.java`field and setter/getters
- [ ] update admin script `sechub-api.sh` inside developer tools project
- [ ] fix test data (if unit tests are failing)
- [ ] update rest doc tests

### PR DESCRIPTION
**Pull request by zigfridus**
This PR will close #1139

### CODE REVIEW

File: sechub-shared-kernel/src/main/java/com/mercedesbenz/sechub/sharedkernel/messaging/UserMessage.java [hunk 1]
```diff
@@ -135,8 +135,8 @@ public String getFormerEmailAddress() {
 
     /**
      * Set the former email address of an user. Should only be called for user
-     * events when an email address has changed. The {@link #getEmailAdress()} shall
-     * contain the new mail adress in this case.
+     * events when an email address has changed. The {@link #getEmailAddress()}
+     * shall contain the new mail address in this case.
```
Comment (Jeeppler) [inline sechub-shared-kernel/src/main/java/com/mercedesbenz/sechub/sharedkernel/messaging/UserMessage.java:139]:
```suggestion
     * shall contain the new email address in this case.
```

File: sechub-administration/src/main/java/com/mercedesbenz/sechub/domain/administration/user/UserCreationService.java [hunk 2]
```diff
@@ -81,14 +81,14 @@ public void createUserFromSelfRegistration(String userId) {
...
+        String emailAddress = selfRegistration.get().getEmailAddress();
+        assertion.assertIsValidEmailAddress(emailAddress);
 
-        found = userRepository.findByEmailAdress(emailAdress);
+        found = userRepository.findByEmailAddress(emailAddress);
 
         if (found.isPresent()) {
-            LOG.warn("Self registration coming in for user:{} but mailadress {} already exists. So just removing self registration entry", sanitizedLogUserId,
-                    emailAdress);
+            LOG.warn("Self registration coming in for user:{} but mailaddress {} already exists. So just removing self registration entry", sanitizedLogUserId,
```
Comment (Jeeppler) [inline sechub-administration/src/main/java/com/mercedesbenz/sechub/domain/administration/user/UserCreationService.java:90]:
```suggestion
            LOG.warn("Self registration coming in for user:{} but email address {} already exists. So just removing self registration entry", sanitizedLogUserId,
```

**Review by Jeeppler — CHANGES_REQUESTED** [review]
Looks good 👍 . I found two minor issues, please fix those.

File: sechub-shared-kernel/src/main/java/com/mercedesbenz/sechub/sharedkernel/messaging/UserMessage.java (hunk 1 above)
Comment (zigfridus) [reply sechub-shared-kernel/src/main/java/com/mercedesbenz/sechub/sharedkernel/messaging/UserMessage.java:139]: Done.

File: sechub-administration/src/main/java/com/mercedesbenz/sechub/domain/administration/user/UserCreationService.java (hunk 2 above)
Comment (zigfridus) [reply sechub-administration/src/main/java/com/mercedesbenz/sechub/domain/administration/user/UserCreationService.java:90]: Done.

**Comment by Jeeppler** [comment]
Please request a re-review or let me know once I should look over the pull-request again.

**Comment by zigfridus** [comment]
@Jeeppler 
I will let you know as soon as I complete this issue.

File: sechub-shared-kernel/src/test/java/com/mercedesbenz/sechub/sharedkernel/validation/UserInputAssertionTest.java [hunk 3]
```diff
@@ -24,17 +24,17 @@ public void before() {
     @Test
     public void only_emailvalidation_is_used_when_email_is_asserted() {
         /* prepare */
-        String validMailAdress = "mymail@example.com";
+        String validMailAddress = "mymail@example.com";
```
Comment (Jeeppler) [inline sechub-shared-kernel/src/test/java/com/mercedesbenz/sechub/sharedkernel/validation/UserInputAssertionTest.java:27]:
```suggestion
        String validEmailAddress = "mymail@example.com";
```

**Review by Jeeppler — CHANGES_REQUESTED** [review]
See comments.

File: sechub-doc/src/test/java/com/mercedesbenz/sechub/restdoc/UserAdministrationRestControllerRestDocTest.java [hunk 4]
```diff
@@ -368,7 +368,7 @@ public void restdoc_show_user_details() throws Exception {
         	                            ),
         	                            responseFields(
         	                                    fieldWithPath(UserDetailInformation.PROPERTY_USERNAME).description("The name of the user"),
-        	                                    fieldWithPath(UserDetailInformation.PROPERTY_EMAIL).description("The mail adress of the user"),
+        	                                    fieldWithPath(UserDetailInformation.PROPERTY_EMAIL).description("The mail address of the user"),
```
[the remark this answers is not in the document]
Comment (zigfridus) [reply sechub-doc/src/test/java/com/mercedesbenz/sechub/restdoc/UserAdministrationRestControllerRestDocTest.java:371]: Done.

File: sechub-administration/src/test/java/com/mercedesbenz/sechub/domain/administration/user/UserEmailAddressUpdateServiceTest.java [hunk 5]
```diff
@@ -138,7 +138,7 @@ void when_assertions_do_not_handle_null_email_user_repository_would_be_called_wi
     }
 
     @Test
-    void throws_not_acceptable_when_same_mail_adress_as_before() {
+    void throws_not_acceptable_when_same_mail_address_as_before() {
```
[the remark this answers is not in the document]
Comment (zigfridus) [reply sechub-administration/src/test/java/com/mercedesbenz/sechub/domain/administration/user/UserEmailAddressUpdateServiceTest.java:141]: Done

File: sechub-doc/src/test/java/com/mercedesbenz/sechub/restdoc/AnonymousUserRequestsNewApiTokenRestDocTest.java [hunk 6]
```diff
@@ -64,7 +64,7 @@ public void calling_with_api_1_0_and_valid_userid_and_email_returns_HTTP_200() t
         /* execute */
         /* @formatter:off */
         this.mockMvc.perform(
-        		post(apiEndpoint,"emailAdress@test.com").
+        		post(apiEndpoint,"emailAddress@test.com").
```
[the remark this answers is not in the document]
Comment (zigfridus) [reply sechub-doc/src/test/java/com/mercedesbenz/sechub/restdoc/AnonymousUserRequestsNewApiTokenRestDocTest.java:67]: Done.

File: sechub-integrationtest/src/test/java/com/mercedesbenz/sechub/integrationtest/scenario2/UserRequestsNewAPITokenIntTest.java [hunk 7]
```diff
@@ -16,7 +16,7 @@ public class UserRequestsNewAPITokenIntTest {
     public IntegrationTestSetup setup = IntegrationTestSetup.forScenario(Scenario2.class);
 
     @Test
-    public void an_anonymous_user_can_trigger_new_api_token_request_for_existing_user_email_adress_leads_to_mail_with_link_to_fetch_new_apitoken() {
+    public void an_anonymous_user_can_trigger_new_api_token_request_for_existing_user_email_address_leads_to_mail_with_link_to_fetch_new_apitoken() {
```
[the remark this answers is not in the document]
Comment (zigfridus) [reply sechub-integrationtest/src/test/java/com/mercedesbenz/sechub/integrationtest/scenario2/UserRequestsNewAPITokenIntTest.java:19]: Done.

File: sechub-notification/src/main/java/com/mercedesbenz/sechub/domain/notification/NotificationConfiguration.java [hunk 8]
```diff
@@ -15,15 +15,15 @@
 @Component
 public class NotificationConfiguration {
 
-    @MustBeDocumented(value = "Single mail adress used for emails to administrators. This should an NPM (non personalized mailbox)")
+    @MustBeDocumented(value = "Single mail address used for emails to administrators. This should an NPM (non personalized mailbox)")
```
[the remark this answers is not in the document]
Comment (zigfridus) [reply sechub-notification/src/main/java/com/mercedesbenz/sechub/domain/notification/NotificationConfiguration.java:18]: Done.

File: sechub-notification/src/test/java/com/mercedesbenz/sechub/domain/notification/user/UserEmailAddressChangedNotificationServiceTest.java [hunk 9]
```diff
@@ -35,13 +35,13 @@ void beforeEach() {
...
-        String formerEmailAddress = "former_email.adress@example.org";
+        String emailAddress = "email.address@example.org";
+        String formerEmailAddress = "former_email.address@example.org";
 
         UserMessage userMessage = new UserMessage();
-        userMessage.setEmailAdress(emailAddress);
+        userMessage.setEmailAddress(emailAddress);
         userMessage.setFormerEmailAddress(formerEmailAddress);
-        userMessage.setSubject("Your mail adress has changed by a test");
+        userMessage.setSubject("Your mail address has changed by a test");
```
[the remark this answers is not in the document]
Comment (zigfridus) [reply sechub-notification/src/test/java/com/mercedesbenz/sechub/domain/notification/user/UserEmailAddressChangedNotificationServiceTest.java:44]: Done.

File: sechub-notification/src/test/java/com/mercedesbenz/sechub/domain/notification/user/UserEmailAddressChangedNotificationServiceTest.java [hunk 10]
```diff
@@ -68,29 +68,29 @@ void sends_emails_to_former_and_new_mail_address_containing_expected_content() {
         String receivedFormerText = mails.receivedFormer.getText();
         String receivedNewText = mails.receivedNew.getText();
 
-        assertEquals("Your mail adress has changed by a test and it will not be used any longer for SecHub.\n"
+        assertEquals("Your mail address has changed by a test and it will not be used any longer for SecHub.\n"
```
[the remark this answers is not in the document]
Comment (zigfridus) [reply sechub-notification/src/test/java/com/mercedesbenz/sechub/domain/notification/user/UserEmailAddressChangedNotificationServiceTest.java:71]: Done.

File: sechub-notification/src/test/java/com/mercedesbenz/sechub/domain/notification/user/UserEmailAddressChangedNotificationServiceTest.java [hunk 11]
```diff
@@ -68,29 +68,29 @@ void sends_emails_to_former_and_new_mail_address_containing_expected_content() {
...
                 + "\n"
                 + "In case you do not receive a follow up notification to the new email address, please inform your SecHub administrator!",
                 receivedFormerText);
-        assertEquals("Your mail adress has changed by a test from former_email.adress@example.org to email.adress@example.org. \n"
+        assertEquals("Your mail address has changed by a test from former_email.address@example.org to email.address@example.org. \n"
```
[the remark this answers is not in the document]
Comment (zigfridus) [reply sechub-notification/src/test/java/com/mercedesbenz/sechub/domain/notification/user/UserEmailAddressChangedNotificationServiceTest.java:75]: Done.

File: sechub-shared-kernel/src/test/java/com/mercedesbenz/sechub/sharedkernel/validation/UserInputAssertionTest.java (hunk 3 above)
Comment (zigfridus) [reply sechub-shared-kernel/src/test/java/com/mercedesbenz/sechub/sharedkernel/validation/UserInputAssertionTest.java:27]: Done.

**Comment by zigfridus** [comment]
@Jeeppler 
I fixed typos from your comments. Please have a look.

**Review by Jeeppler — APPROVED** [review]
Looks good. Thanks for your contribution 🥇

**Comment by Jeeppler** [comment]
@zigfridus can you fix the merge request? Simply merge the community branch into your branch and overwrite the file (merge it). Since the `reduced-openapi3.json` is auto generated, it does not matter if you use the upstream version or your version. It is only important, that this file exists.

**Comment by zigfridus** [comment]
@Jeeppler 
My branch had been created before `community` branch and was  connect to the `develop` branch. Could you please advise how to correctly merge it in my case? Thanks
image

**Comment by Jeeppler** [comment]
@zigfridus give the following a try:

1. Checkout the `community` branch
2. Take the `sechub-api-java/src/main/resources/reduced-openapi3.json` JSON and copy it to any folder outside of the sechub git repository
3. Switch to your branch to `feature-1139-rename-parts-with-emailAdress` 
4. copy & replace the `sechub-api-java/src/main/resources/reduced-openapi3.json` with the file on your branch
5. Push the changes (important: don't build your branch)

If the `sechub-api-java/src/main/resources/reduced-openapi3.json` file is identical on the community branch and your branch the merge conflict should be resolved. 

NOTE: Don't build your branch again. Otherwise the `reduced-openapi3.json` will be generated again.

**Comment by zigfridus** [comment]
@Jeeppler 
Finally no conflicts. Thank you for the advice!

**Pull request merged**